src/schemas/text-based-fields.ts

```diff
@@ -95,13 +101,16 @@ const generateValidationSchema = (elementType: EElementType) => {
                                     ),
                         })
                         .when("validationType", {
-                            is:
-                                ELEMENT_VALIDATION_TYPES["Text field"][
-                                    EElementType.TEXT
-                                ].validationTypes[1] ||
-                                ELEMENT_VALIDATION_TYPES["Text field"][
-                                    EElementType.TEXT
-                                ].validationTypes[2],
+                            is: (value: string) => {
+                                const validationTypes =
+                                    ELEMENT_VALIDATION_TYPES["Text field"][
+                                        EElementType.TEXT
+                                    ].validationTypes;
+                                return (
+                                    validationTypes.includes(value) &&
+                                    value !== "Custom regex"
+                                );
```

Comment:
apologies, on second look I think it's better to handle each validation type explicitly

also, remove the `otherwise` handler because it will never run

```suggestion
                                return (
                                    validationTypes.includes(value) &&
                                    (value === "Minimum length" ||
                                        value === "Maximum length")
                                );
```